src/pages/AddVacation/AddVacation.tsx

```diff
@@ -1,30 +1,135 @@
-import React, { useState } from "react";
+import React, { useState, useContext, useEffect } from "react";
 import { Dayjs } from "dayjs";
 import { useNavigate } from "react-router-dom";
+import { v4 as uuidv4 } from "uuid";
 import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
 import TextField from "@mui/material/TextField";
 import { VacationDatePicker } from "./DatePicker";
 import { VacationSelector } from "./VacationSelector";
 import { Layout } from "../../shared/Layout";
 import { styledAddVacation } from "./style";
+import { IDataVacation, UserData } from "../../context/UserDataContext";
+import { useAuth } from "../../hooks/useAuth";
 
 const AddVacation = () => {
   const [vacationType, setVacationType] = useState<string>("");
   const [startDate, setStartDate] = useState<Dayjs | null>(null);
   const [endDate, setEndDate] = useState<Dayjs | null>(null);
   const [notes, setNotes] = useState<string>("");
   const navigate = useNavigate();
+  const { data, setData, edit, setEdit } = useContext(UserData);
+  const { user } = useAuth();
 
   const isFieldsEmpty = startDate && endDate && vacationType;
+  const setHeaderTitle: string = !edit ? "Add vacation" : "Edit vacation";
 
-  const addNewVacation = () => {
-    if (!isFieldsEmpty) return console.log("Error");
-    console.log({ vacationType, endDate, startDate });
+  useEffect(() => {
+    if (edit) {
+      const editVacation = data.find((vacation) => vacation.id === edit);
```

Comment:
you shouldn't store editID in the context at all. When a user clicks on a dashboard edit vacation, you can add ID as a query param in the URL and use it here, in this component to select the proper data set for Data, provided by the Context. If the query param is undefined, you might assume that it's a new vacation page state.